### PR TITLE
Added OrderStatusComboBox implementation

### PR DESCRIPTION
**Pull request by spireken**
Demo is called OrderStatusComboBox.

Would it make sense to insert the values manually, so they can be put in alphabetical order?

### CODE REVIEW

File: Applications/Spire/Source/Ui/OrderStatusComboBox.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#include "Spire/Ui/OrderStatusComboBox.hpp"
+#include <QHBoxLayout>
+#include "Beam/Collections/EnumIterator.hpp"
```
Comment (kranar) [inline Applications/Spire/Source/Ui/OrderStatusComboBox.cpp:3]: Treat Beam as a third party library, use ```<...>``` to include it.

Comment (spireken) [reply Applications/Spire/Source/Ui/OrderStatusComboBox.cpp:3]: So alphabetically it now is included before <QHBoxLayout>?

Comment (kranar) [reply Applications/Spire/Source/Ui/OrderStatusComboBox.cpp:3]: Included before what?

Comment (spireken) [reply Applications/Spire/Source/Ui/OrderStatusComboBox.cpp:3]: Mean to say that would `<QHBoxLayout>` be included after Beam includes?

**Pull request merged**